kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NetworkPropertiesTest.java

```diff
@@ -0,0 +1,231 @@
+		givenNetworkPropsIsCreated();
+		whenGetOptIsCalledWith("", String.class);
+		thenResultEquals(Optional.empty());
+	}
+
+	@Test
+	public void shouldReturnStringListGetStringListWithCommaSeperatedString() {
+		givenTheMapWith("testKey-comma-seperated", "commaSeperated1,commaSeperated2,commaSeperated3");
+		givenNetworkPropsIsCreated();
+		whenGetStringListIsCalledWith("testKey-comma-seperated");
+		thenResultEquals(List.of("commaSeperated1", "commaSeperated2", "commaSeperated3"));
+	}
+
+	@Test
+	public void shouldReturnStringListGetStringListWithCommaSeperatedStringMalformed() {
+		givenTheMapWith("testKey-comma-seperated",
+				",,   ,,,commaSeperated1, ,,,,commaSeperated2,   ,,commaSeperated3,");
+		givenNetworkPropsIsCreated();
+		whenGetStringListIsCalledWith("testKey-comma-seperated");
+		thenResultEquals(List.of("commaSeperated1", "commaSeperated2", "commaSeperated3"));
+	}
+
+	@Test
+	public void shouldReturnStringListGetStringListWithRegularString() {
+		givenTheMapWith("testKey1", "testString1");
+		givenNetworkPropsIsCreated();
+		whenGetStringListIsCalledWith("testKey1");
+		thenResultEquals(List.of("testString1"));
+	}
+
+	@Test
+	public void shouldReturnStringListGetStringListWithNull() {
+		givenTheMapWith("testKeyNull", null);
+		givenNetworkPropsIsCreated();
+		whenGetStringListIsCalledWith("testKeyNull");
+		thenResultEquals(List.of());
+	}
+
+	@Test
+	public void shouldReturnOptionalStringListGetStringListOptWithCommaSeperatedString() {
+		givenTheMapWith("testKey-comma-seperated", "commaSeperated1,commaSeperated2,commaSeperated3");
+		givenNetworkPropsIsCreated();
+		whenGetStringListOptIsCalledWith("testKey-comma-seperated");
+		thenResultEquals(Optional.of(List.of("commaSeperated1", "commaSeperated2", "commaSeperated3")));
+	}
+
+	@Test
+	public void shouldReturnOptionalStringListGetStringListOptWithCommaSeperatedStringMalformed() {
+		givenTheMapWith("testKey-comma-seperated",
+				", , ,,,,commaSeperated1, , , ,,,,,commaSeperated2,,,, ,, ,,commaSeperated3,, , ,,,, ,");
+		givenNetworkPropsIsCreated();
+		whenGetStringListOptIsCalledWith("testKey-comma-seperated");
+		thenResultEquals(Optional.of(List.of("commaSeperated1", "commaSeperated2", "commaSeperated3")));
+	}
+
+	@Test
+	public void shouldReturnOptionalStringListGetStringListOptWithString() {
+		givenTheMapWith("testKey1", "testString1");
+		givenNetworkPropsIsCreated();
+		whenGetStringListOptIsCalledWith("testKey1");
+		thenResultEquals(Optional.of(List.of("testString1")));
+	}
+
+	@Test
+	public void shouldReturnEmptyGetStringListOptWithNull() {
+		givenTheMapWith("testKeyNull", null);
+		givenNetworkPropsIsCreated();
+		whenGetStringListOptIsCalledWith("testKeyNull");
+		thenResultEquals(Optional.empty());
+	}
+
+	@Test
+	public void shouldReturnEmptyGetStringListOptWithNoKey() {
+		givenTheMapWith("testKeyNull", null);
+		givenNetworkPropsIsCreated();
+		whenGetStringListOptIsCalledWith("");
+		thenResultEquals(Optional.empty());
+	}
+
+	// Given //
```

Comment:
Finally (and this is my OCD that's talking) in the past we used the following comment style for splitting the Gerkhin-style sections:

```suggestion
     /*
     * Given
     */
```